src/engine/Filter.cpp

```diff
@@ -37,10 +39,34 @@ string Filter::getCacheKeyImpl() const {
   return std::move(os).str();
 }
 
+//______________________________________________________________________________
 string Filter::getDescriptor() const {
   return absl::StrCat("Filter ", _expression.getDescriptor());
 }
 
+//______________________________________________________________________________
+void Filter::setPrefilterExpressionForDirectIndexScanChild() {
+  std::vector<PrefilterVariablePair> prefilterPairs =
+      _expression.getPrefilterExpressionForMetadata();
+  std::vector<PrefilterVariablePair> relevantPairs;
+  relevantPairs.reserve(prefilterPairs.size());
+  VariableToColumnMap varToColMap = _subtree->getVariableColumns();
+  // Add all the PrefilterVariable values whose Variable value is
+  // contained in the VariableToColumnMap. This is done to avoid that certain
+  // subqueries filter out too much.
+  for (auto& prefilterPair : prefilterPairs) {
+    if (varToColMap.find(prefilterPair.second) != varToColMap.end()) {
+      relevantPairs.emplace_back(std::move(prefilterPair));
+    }
+  }
```

Comment:
The filtering by the visible variables can be done by the `QueryExecutionTree` class.
(`QueryExecutionTree::setPrefilterExpr...)`